### DISCUSSION TITLE
system.reflection.reflectiontypeloadexception error dll

### DISCUSSION THREAD

**Discussion by Pgabrielmnecraft** [TweakScale/TweakScale#259] [Support] [answered]
I was trying to start my ksp for a while, but whenever I start this error appears, I don't know what else to do.
[KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/9033562/KSP.log)

@lisias

**Comment by Pgabrielmnecraft** [comment]
@lisias help pls

**Reply by Lisias** [reply]
Looking on it. Give me half an hour.

**Comment by Lisias** [comment, marked as answer]
Hi, @Pgabrielmnecraft . Found the problem.

You was bitten by a nasty KSP bug on a thingy called Assembly.Resolver. It's the thing that loads DLLs from your GameData. The bug is that once a DLL fails to be loaded (by any reason, as a faulty dependency or being compiled by something Mono does not understand), **everything else** fails to be loaded later, no matter being fine or not.

What happens is that TweakScale, aware of the problem, yells when it doesn't find its own DLLs, so you can do something about instead of loading the savegame and being screwed up by loading lots of things with missing PartModules.

So, usually, the first occurrence of the `TypeLoadException` on the KSP.log tells us what really is at fault on your rig (and everything else is that bug on the Assembly.Resolver being triggered).

On your rig, this is:

```
[ERR 15:32:10.815] ADDON BINDER: Cannot resolve assembly: 0Harmony, Culture=neutral, PublicKeyToken=null

[ERR 15:32:10.815] ADDON BINDER: Cannot resolve assembly: 0Harmony, Culture=neutral, PublicKeyToken=null

[EXC 15:32:10.819] TypeLoadException: Could not load type of field 'LmpClient.Base.HarmonyPatcher:HarmonyInstance' (0) due to: Could not load file or assembly
 '0Harmony, Version=2.0.4.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. assembly:0Harmony, Version=2.0.4.0, Culture=neutral, PublicKeyT
oken=null type:<unknown type> member:(null) signature:<none>
        UnityEngine.DebugLogHandler:LogException(Exception, Object)
        ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
        UnityEngine.GameObject:AddComponent(Type)
        AddonLoader:StartAddon(LoadedAssembly, Type, KSPAddon, Startup)
        AddonLoader:StartAddons(Startup)
        <LoadObjects>d__90:MoveNext()
        UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
        <CreateDatabase>d__71:MoveNext()
        UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
        GameDatabase:StartLoad()
        <LoadSystems>d__11:MoveNext()
        UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
        LoadingScreen:Start()
```

You have something that needs Harmony installed. Install it and you should be able to play your game.

I want o emphasise that TweakScale **does not** uses Harmony, and it is not at fault here. TweakScale is the one yelling about a problem before it bites you in the SAS.

Cheers!

**Comment by Lisias** [comment]
Assuming the ticket is concluded with success. Closing this.
